--- a/libavdevice/v4l2.c
+++ b/libavdevice/v4l2.c
@@ -60,7 +60,7 @@ typedef struct V4L2Context {
     AVClass *class;
     int fd;
     int ioctl_retry;
-    int (*ioctl_f)(int, int, ...);
+    int (*ioctl_f)(int, unsigned long int, ...);
     int (*open_f)(const char *, int, ...);
     int (*close_f)(int);
     ssize_t (*read_f)(int, void *, size_t);
@@ -134,7 +134,7 @@ static int device_open(AVFormatContext *ctx, const char *device)
 {
     V4L2Context *s = ctx->priv_data;
     struct v4l2_capability cap;
-    int (*ioctl_f)(int, int, ...);
+    int (*ioctl_f)(int, unsigned long int, ...);
     int fd = -1;
 
     s->ioctl_retry = 0;
